Azure-Samples/chat-with-your-data-solution-accelerator
# feat: Generate and add image captions to search index when image is ingested.

Closes #749

## Purpose
This PR adds the following functionality when an image is uploaded to the knowledge base (when advanced image processing is enabled):
* generates a caption for the image by calling the OpenAI service
* generates the vector embeddings for the caption
* includes the caption text and embeddings in the document uploaded to the search index

This will allow for improved search on images.

## Does this introduce a breaking change?



## How to Test
Deploy Chat with your data with Advanced Image Processing enabled.
Upload an image via the admin site. 
Then verify on the search index that the `content` and `content_vector` fields are populated with the appropriate caption and vector embeddings.
